addons/point_of_sale/migrations/13.0.1.0.1/post-migration.py

```diff
@@ -0,0 +1,185 @@
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE pos_order po
+        SET account_move = am.id
+        FROM account_invoice ai
+        JOIN account_move am ON am.old_invoice_id = ai.id
+        WHERE po.{} = ai.id
+        """.format(openupgrade.get_legacy_name('invoice_id'))
+    )
+
+
+def fill_pos_config_barcode_nomenclature_id(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE pos_config pc
+        SET barcode_nomenclature_id = rc.nomenclature_id
+        FROM res_company rc
+        WHERE pc.company_id = rc.id AND pc.barcode_nomenclature_id IS NULL
+        """,
+    )
+
+
+def fill_pos_config_module_pos_hr(env):
+    if openupgrade.table_exists(env.cr, 'hr_employee'):
+        openupgrade.logged_query(
+            env.cr, """
+            UPDATE pos_config SET module_pos_hr = TRUE""",
+        )
+
+
+def delete_pos_order_line_with_empty_order_id(env):
+    openupgrade.logged_query(
+        env.cr, "DELETE FROM pos_order_line WHERE order_id IS NULL",
+    )
+
+
+def fill_stock_warehouse_pos_type_id(env):
+    warehouses = env['stock.warehouse'].search([('pos_type_id', '=', False)])
+    for warehouse in warehouses:
+        all_used_colors = [res['color'] for res in env[
+            'stock.picking.type'].search_read(
+            [('warehouse_id', '!=', False), ('color', '!=', False)],
+            ['color'], order='color')]
+        available_colors = [
+            zef for zef in range(0, 12) if zef not in all_used_colors]
+        color = available_colors[0] if available_colors else 0
+        sequence = env['ir.sequence'].sudo().create({
+            'name': warehouse.name + ' ' + 'Picking POS',
+            'prefix': warehouse.code + '/POS/',
+            'padding': 5,
+            'company_id': warehouse.company_id.id,
+        })
+        pos_type = env['stock.picking.type'].create({
+            'name': 'PoS Orders',
+            'code': 'outgoing',
+            'default_location_src_id': warehouse.lot_stock_id.id,
+            'default_location_dest_id': env.ref(
+                'stock.stock_location_customers').id,
+            'sequence': sequence.id,
+            'sequence_code': 'POS',
+            'company_id': warehouse.company_id.id,
+            'warehouse_id': warehouse.id,
+            'color': color,
+        })
+        warehouse.write({'pos_type_id': pos_type.id})
+
+
+def create_pos_payment_methods(env):
+    env.cr.execute("""
+        SELECT DISTINCT journal_id
+        FROM pos_config_journal_rel
+    """)
+    journal_ids = [aj[0] for aj in env.cr.fetchall()]
+    if journal_ids:
+        vals_list = []
+        for journal in env['account.journal'].browse(journal_ids):
+            vals = {
+                'name': 'Cash',
+                'cash_journal_id': journal.id,
+                'is_cash_count': True,
+                'company_id': journal.company_id.id,
+            }
+            vals_list += [vals]
+        env['pos.payment.method'].create(vals_list)
```

Comment:
I have an issue here because `pos.payment.method.receivable_account_id` is required and it's default value comes from a new field `res.company.account_default_pos_receivable_account_id`.

Reply:
moved method to end-migration because `res.company.account_default_pos_receivable_account_id` is filled in end-migration of `account`.

Reply:
Please check this that I have fixed all of this:

https://github.com/Tecnativa/OpenUpgrade/commit/146a30fe48197ac0c190039efdf03cc471df5245
https://github.com/Tecnativa/OpenUpgrade/commit/238da181379def6b8f8f12aa0932aa16f248b027